commands/util.go

```diff
@@ -475,3 +476,31 @@ func boot(ctx context.Context, ngi *nginfo) (bool, error) {
 
 	return true, err
 }
+
+func printInstance(instance string, dockerCli command.Cli, l progress.Logger) error {
+	var ng *store.NodeGroup
+	var err error
+	txn, release, err := storeutil.GetStore(dockerCli)
+	if err != nil {
+		return err
+	}
+	defer release()
+	if instance != "" {
+		ng, err = storeutil.GetNodeGroup(txn, dockerCli, instance)
+		if err != nil {
+			return err
+		}
+	} else {
+		ng, err = storeutil.GetCurrentInstance(txn, dockerCli)
+		if err != nil {
+			return err
+		}
+	}
+	if ng.Driver == "" {
+		ng.Driver = "docker"
+	}
+	return progress.Wrap(fmt.Sprintf("[internal] building with %q instance using %s driver", ng.Name, ng.Driver), l, func(sub progress.SubLogger) error {
+		// noop
+		return nil
```

Comment:
I think we should be able to do this with a `oneOffProgress` function: https://github.com/moby/buildkit/blob/master/exporter/containerimage/writer.go#L561-L576 to be more consistent with how we already do these kind of one-off messages.

Although, I'm not sure why we have so many identical private `oneOffProgress` functions, we should definitely be able to refactor all of them together.